include/Mat.h

```diff
@@ -1067,3 +1103,69 @@ template<class Type>
 Mat<bool> operator>=(Type a, Mat<Type> &b){
     return Mat<Type>::broadcast(a, b, GreaterThanEqual<Type>);
 }
+template<class Type>
+Mat<Type> Mat<Type>::imask(Mat<bool> &mask){
+    errorCheck(mask.rows() != rows(),
+        "Matrix shape mismatch in call to i()");
+    errorCheck(ndims == 2 && mask.columns() != columns(),
+        "Matrix shape mismatch in call to i()");
+    size_type newSize = 0;
+    for(auto i : mask){
+        if(i) newSize++;
+    }
+    Mat<Type> out(newSize);
+
+    iterator j = begin();
+    Mat<bool>::iterator k = mask.begin();
+    for(auto& i : out){
+        while(*k == false){ j++; k++; }
+        i = *j;
+        j++; k++;
+    }
+    return out;
+}
+template<class Type>
+template<class Type2>
+Mat<Type> Mat<Type>::inum(Mat<Type2> &indices){
+    if(ndims == 2){
+        Mat<Type> out(indices.size(), columns());
+        for(size_type i = 0; i < indices.size(); i++){
+            for(size_type j = 0; j < columns(); j++){
+                out(i,j) = operator()(indices(i),j);
+            }
+        }
+        return out;
+    }
+    else if(ndims == 1){
+        Mat<Type> out(indices.size());
+        for(size_type i = 0; i < indices.columns(); i++){
+            out(i) = operator()(indices(i));
+        }
+        return out;
+    }
+    else{
+        errorCheck(true, "n-dimensional fancy indexing not yet implemented");
+        return Mat<Type>::zeros(0);
+    }
+}
+template<class Type>
+void Mat<Type>::itomask(Mat<bool> &mask, Mat<Type> &out){
+    errorCheck(mask.rows() != rows(),
+        "Matrix shape mismatch in call to i()");
+    errorCheck(ndims == 2 && mask.columns() != columns(),
+        "Matrix shape mismatch in call to i()");
+    Mat<Type>::size_type newSize = 0;
+    for(auto i : mask){
+        if(i) newSize++;
+    }
+    Mat<Type> temp(newSize);
```

Comment:
- If output is not 1D, error.
- Error if output is too small. 
- If output is too big, use roi.
- If right size, just copy